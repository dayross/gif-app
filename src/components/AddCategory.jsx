import {React, useState} from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({onNewCategory}) => {
    const [inputValue, setInputValue] = useState([]);

    const onInputChange = (event) =>{
        setInputValue(event.target.value, ...inputValue);
    };

    const onSubmit = (event) => {
        event.preventDefault();
        if(inputValue.trim().length<=1) return;
        // lamar el set  ategories y desestructurarlo y mandar el input value
        // setCategories(categories => [inputValue, ...categories]);
        onNewCategory(inputValue.trim());
        setInputValue('');
    };
    return (

        <>
            <form aria-label="form" onSubmit={(event) => onSubmit(event)}>
                <input type="text"
                placeholder="Search"
                // el valor que guarda llevara ese nombre
                value = {inputValue}
                // y cada que cambie se hara un trigger a la funcion onInpChange
                onChange = { (event) => onInputChange(event)}
                />
            </form>

            
            <p />
        </>
    );
};

AddCategory.propTypes = {
    onNewCategory : PropTypes.func.isRequired
}