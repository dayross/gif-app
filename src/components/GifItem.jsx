import React from 'react';
import PropTypes from 'prop-types';

export const GifItem = ({title, url}) => {
    
  return (

    <div> 
      <img src={url} alt={title}/>
      <p className="card"> {title} </p>
    </div>
    
  )
}

GifItem.propTypes = {
  title : PropTypes.string.isRequired,
  url : PropTypes.string.isRequired
}