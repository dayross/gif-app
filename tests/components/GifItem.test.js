import { fireEvent, render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem";

describe('Set de pruebas para GifItem.js', ()=>{

    const title = 'zoologico';
    const url = 'http://asdfghjkl.com/';

    // test('Probando que los tipos de datos sean los correctos', ()=>{
    //     const {title, url} = render(<GifItem title="hola" url="person.com"/>);
    //     expect(typeof (title)).toEqual('string');
    // });


    test('Hacer match con el snapshot', ()=>{
        const {container} = render(<GifItem title="hola" url="person.com"/>);
        expect(container).toMatchSnapshot();
    });

    test('Debe mostrar la imagen con el url y titulo indicado', ()=>{
        render(<GifItem title= {title} url={url} />);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('Debe mostrar el titulo en pantalla', ()=>{
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();

    });
});