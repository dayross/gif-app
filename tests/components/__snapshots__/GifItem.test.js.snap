// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Set de pruebas para GifItem.js Hacer match con el snapshot 1`] = `
<div>
  <div>
    <img
      alt="hola"
      src="person.com"
    />
    <p
      class="card"
    >
       
      hola
       
    </p>
  </div>
</div>
`;
